airsonic-main/src/main/java/org/airsonic/player/service/upnp/AlbumUpnpProcessor.java

```diff
@@ -61,20 +59,6 @@ public AlbumUpnpProcessor() {
         setRootTitle("Albums");
     }
 
-    /**
-     * Browses the top-level content of a type.
-     */
-    public BrowseResult browseRoot(String filter, long firstResult, long maxResults, SortCriterion[] orderBy) throws Exception {
-        DIDLContent didl = new DIDLContent();
-
-        List<MusicFolder> allFolders = getDispatchingContentDirectory().getSettingsService().getAllMusicFolders();
-        List<Album> selectedItems = getAlbumDao().getAlphabeticalAlbums((int) firstResult, (int) maxResults, false, true, allFolders);
-        for (Album item : selectedItems) {
-            addItem(didl, item);
-        }
-
-        return createBrowseResult(didl, (int) didl.getCount(), getAllItemsSize());
-    }
```

Comment:
Just to make sure I understand, the above is just a code cleanup as it does nothing functionally different than the base class, correct?

Reply:
Oh wait--hold off on this. Now I know what happened; I had thought that for some weird reason I'd left in some old un-refactored code which was breaking the RecentAlbumUpnpProcessor. But now that I'm looking more closely I realize that this was actually an optimization that had been put in...which was breaking the RecentAlbumUpnpProcessor.

This fix just removes the optimization, which works but isn't, erm, optimal.  The right thing to do is to move the optimization so it doesn't break its subclasses.